### PR TITLE
Update tools to ensure error details appear in VS

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Partially addresses https://github.com/grpc/grpc-dotnet/issues/129.

This PR enables the ProtoCompile task to output errors which will then be picked up by the VS error list. 

Before:
image

After:
image

### CODE REVIEW

File: src/csharp/Grpc.Tools/ProtoCompile.cs [hunk 1]
```diff
@@ -322,6 +322,7 @@ protected override string GenerateResponseFileCommands()
             {
                 cmd.AddArg(proto.ItemSpec);
             }
+            cmd.AddSwitchMaybe("error_format", "msvs");
```
Comment (JunTaoLuo) [inline src/csharp/Grpc.Tools/ProtoCompile.cs]: @jtattermusch where is the source code for protoc.exe? I see it being downloaded from maven: https://github.com/protocolbuffers/protobuf/blob/master/csharp/build_tools.sh#L43 but I don't know where the source code lives. It looks like the error format is not correct so the Error list in VS cannot link to the correct location in code.

**Comment by JunTaoLuo** [comment]
cc @jtattermusch @JamesNK @shirhatti

**Comment by JunTaoLuo** [comment]
WIP need to fix tests.

File: src/csharp/Grpc.Tools/ProtoCompile.cs (hunk 1 above)
Comment (jtattermusch) [reply src/csharp/Grpc.Tools/ProtoCompile.cs]: Sources for protoc.exe are in https://github.com/protocolbuffers/protobuf, but we are actually building protoc.exe ourselves (from the third_party/protobuf git submodule) while building  the native artifacts before we proceed to building the nugets.

https://github.com/grpc/grpc/blob/7095e051f2183b632aac241c27cf6b53b69d50e9/tools/run_tests/artifacts/artifact_targets.py#L301

File: src/csharp/Grpc.Tools/ProtoCompile.cs [hunk 2]
```diff
@@ -318,6 +318,7 @@ protected override string GenerateResponseFileCommands()
                     cmd.AddSwitchMaybe("proto_path", TrimEndSlash(path));
             }
             cmd.AddSwitchMaybe("dependency_out", DependencyOut);
+            cmd.AddSwitchMaybe("error_format", "msvs");
```
Comment (jtattermusch) [inline src/csharp/Grpc.Tools/ProtoCompile.cs]: should we only use msvs  error format when we are on windows?

**Comment by jtattermusch** [comment]
Triggered an adhoc nuget build: https://sponge.corp.google.com/invocation?tab=Build+Log&id=4f864702-597a-433d-83cb-006598e665fc&searchFor= (will post results later).

File: src/csharp/Grpc.Tools/ProtoCompile.cs (hunk 2 above)
Comment (JunTaoLuo) [reply src/csharp/Grpc.Tools/ProtoCompile.cs]: Let's use msvs format even on other platforms so errors on dotnet are consistent. I don't see why dotnet projects on OSX and Linux should behave differently.

File: src/csharp/Grpc.Tools/ProtoCompile.cs (hunk 1 above)
Comment (JunTaoLuo) [reply src/csharp/Grpc.Tools/ProtoCompile.cs]: I'll see if I can find the source and fix the format but looks like it'll probably go into the protobuf repo.

**Comment by kkm000** [comment]
@JunTaoLuo, thanks much! I just checked my early notes from my OneNote notebook, and it simply fell through the cracks. Really appreciate the fix.

@jtattermush, I second this decision to always use MSVS format. It's not a bad format in itself, just a couple extra parens vs gcc, so human console builders will have no trouble associating the message line to source location. Also, C# IDEs (VS Code, for example) probably support it out of the box even when ported. The only thing it can potentially confuse is emacs, but its users usually possess enough emacs-fu to add a regexp to catch them. So I believe is't a correct thing to set this in tooling unconditionally. Looks like I even pondered that, if I decipher that old note to myself (but I have not even a hint of recollection  writing it!).

**Comment by kkm000** [comment]
Hmm, I'm wondering why did not this change break ProtoCompileCommandLineGeneratorTest.cs. Looks like I am doing an exact comparison of command line in the simplest case (i. e. "must be this" vs a more relaxed "must contain this"). I'll check.

See, e.g, https://github.com/grpc/grpc/blob/c9b4d8347/src/csharp/Grpc.Tools.Tests/ProtoCompileCommandLineGeneratorTest.cs#L56 -- `AssertThat(... IsEqualTo...)`.

@JunTaoLuo, it's on my plate, don't worry.

**Comment by JunTaoLuo** [comment]
@kkm000 You might also want to take a look at https://github.com/grpc/grpc/pull/18446 which was the second part of the issue this PR was trying to address. 

The MSVS format actually seems incorrect from the point of view of Visual Studio actually so I'm not sure where that came from, xref https://github.com/protocolbuffers/protobuf/issues/5951.

**Pull request merged**